### ISSUE TITLE
I set the layer.trainable = False but the param still changed after train,why?

### ISSUE THREAD

**Issue by hanzy123** [fchollet/keras#7051]
Please make sure that the boxes below are checked before you submit your issue. If your issue is an implementation question, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [join the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) and ask there instead of filing a GitHub issue.

Thank you!

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [x] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [ ] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [ ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by hanzy123** [comment]
when i was building model : 
```
model = ResNet50(include_top=False, input_tensor=preproc)
mid_start = model.get_layer('res5b_branch2a')
all_layers = model.layers
for i in range(model.layers.index(mid_start)):
    all_layers[i].trainable = False
mid_out = model.layers[model.layers.index(mid_start) - 1]
rn_top = Model(model.input, mid_out.output)
```
then i add my own layers:
```
model = Sequential()
model.add(rn_top)

model1 = AveragePooling2D((7, 7))(model.output)
model1 = identity_block(model1, 3, [512, 512, 2048], stage=5, block='b')
model1 = identity_block(model1, 3, [512, 512, 2048], stage=5, block='c')
model1 = Flatten()(model1)
model1 = Dense(1024)(model1)
model1 = BatchNormalization()(model1)
model1 = Dropout(0.25)(model1)
model1 = Dense(1024)(model1)
model1 = BatchNormalization()(model1)
model1 = Dropout(0.2)(model1)
x_class = Dense(2, activation='softmax', name='class')(model1)

final_model = Model(rn_top.input, x_class)
```

**Comment by hanzy123** [comment]
then i init two new Resnet50 models ,and put final_model.layers[0:155](`res5b_branch2a` is the 155 layer in Resnet50)'s param to one of the models. then predict the output of the `res5b_branch2a` layer get featuremap1 and featuremap2.

finally , i use `numpy.allclose(featuremap1, featuremap2)`, but i get `False`, why?

**Comment by hanzy123** [comment]
And when i run final_model.summary()
i get ：
```
Total params: 26,745,730
Trainable params: 12,085,250
Non-trainable params: 14,660,480
```

**Comment by fchollet** [comment]
Because you are using batch norm, which updates its internal state regardless of trainability status (i.e. it updates its batch statistics, which are non-trainable). This is expected behavior.

**Comment by ahundt** [comment]
Note to those who come across this BatchNormalization now properly locks when trainable=False.
